common/src/main/java/com/revenuecat/purchases/common/Backend.kt

```diff
@@ -114,10 +119,13 @@ class Backend(
                         if (result.isSuccessful()) {
                             onSuccess(CustomerInfoFactory.buildCustomerInfo(result))
```

Comment:
We are building the `CustomerInfo` here. Do we need to send to lower layers that the backend is down? Or could we check here if it's a server error and compute the offline customer info and send it to the `onSuccess`? That way we isolate the logic in the networking layer. We might not want to do that though. Let me know what you think.

Reply:
I started on a possible approach for this logic here: #964. I kinda prefer to keep it outside the networking layer since seems like the logic doesn't really belong there... 

In that PR, I tried to move as much logic as possible to the `OfflineEntitlementsManager`, but there is still logic in the `Purchases` and `CustomerInfoHelper` classes. And there is actually some duplication in `Purchases` due to that... I actually would like to refactor some code there to avoid the duplication, but it's tricky... 

We can discuss this more over Slack